test-zone-data/Address-TP/address03/test-zones-output.md

```diff
@@ -0,0 +1,195 @@
+|:--------------------|:---------------------------|:-----------------------|
+| PTR-IS-GOOD-CNAME-1 | NAMESERVER\_IP\_PTR\_MATCH | 2)                     |
+
+* (2) All tags except for those specified as “Mandatory message tags”
+
+```
+$ zonemaster-cli --raw --show-testcase --test address03 --hints COMMON/hintfile --level info ptr-is-good-cname-1.address03.xa
+   0.00 INFO     Unspecified    GLOBAL_VERSION  version=v7.0.0
+   0.15 INFO     Address03      NAMESERVER_IP_PTR_MATCH
+```
+--> OK
+
+| Scenario name       | Mandatory message tag      | Forbidden message tags |
+|:--------------------|:---------------------------|:-----------------------|
+| PTR-IS-GOOD-CNAME-2 | NAMESERVER\_IP\_PTR\_MATCH | 2)                     |
+
+* (2) All tags except for those specified as “Mandatory message tags”
+
+```
+$ zonemaster-cli --raw --show-testcase --test address03 --hints COMMON/hintfile --level info ptr-is-good-cname-2.address03.xa
+   0.00 INFO     Unspecified    GLOBAL_VERSION  version=v7.0.0
+   0.14 WARNING  Address03      NAMESERVER_IP_WITHOUT_REVERSE  ns_ip=127.11.3.41; nsname=ns11.child.address03.xa
+   0.16 WARNING  Address03      NAMESERVER_IP_WITHOUT_REVERSE  ns_ip=fda1:b2:c3:0:127:11:3:41; nsname=ns11.child.address03.xa
+```
+
+The output is incorrect with Zonemaster-Engine version 7.0.0 due to a bug (<https://github.com/zonemaster/zonemaster/issues/1353>).
+
+--> Not yet OK
+
+| Scenario name         | Mandatory message tag            | Forbidden message tags |
+|:----------------------|:---------------------------------|:-----------------------|
+| PTR-IS-DANGLING-CNAME | NAMESERVER\_IP\_WITHOUT\_REVERSE | 2)                     |
+
+* (2) All tags except for those specified as “Mandatory message tags”
+
+```
+$ zonemaster-cli --raw --show-testcase --test address03 --hints COMMON/hintfile --level info ptr-is-dangling-cname.address03.xa
+   0.00 INFO     Unspecified    GLOBAL_VERSION  version=v7.0.0
+   0.07 WARNING  Address03      NAMESERVER_IP_WITHOUT_REVERSE  ns_ip=127.11.3.42; nsname=ns12.child.address03.xa
+   0.08 WARNING  Address03      NAMESERVER_IP_WITHOUT_REVERSE  ns_ip=fda1:b2:c3:0:127:11:3:42; nsname=ns12.child.address03.xa
+   0.08 WARNING  Address03      NAMESERVER_IP_WITHOUT_REVERSE  ns_ip=127.11.3.35; nsname=ns5.child.address03.xa
+   0.09 WARNING  Address03      NAMESERVER_IP_WITHOUT_REVERSE  ns_ip=fda1:b2:c3:0:127:11:3:35; nsname=ns5.child.address03.xa
+```
+--> OK
+
+| Scenario name        | Mandatory message tag            | Forbidden message tags     |
+|:---------------------|:---------------------------------|:---------------------------|
+| PTR-IS-ILLEGAL-CNAME | NAMESERVER\_IP\_WITHOUT\_REVERSE | NAMESERVER\_IP\_PTR\_MATCH |
+
+Currently, the test case also emits `NO_RESPONSE_PTR_QUERY` as a side-effect of the recursor code aborting the resolution because of the illegal alias. Until the specification is rewritten, we do not mandate nor forbid this message tag.

```

Comment:
I understand, and that answers my question in the specification of the scenarios. I think this explanation belongs to the specification, not to this document.